### DISCUSSION TITLE
Can we subscript a String in mojo?

### DISCUSSION THREAD

**Discussion by abhinav-upadhyay** [modular/modular#95] [Q&A] [answered]
In Python strings, you can do this:

```python
s = 'hello, world'
print(s[0])
```

In mojo, this doesn't work:
```python
let s = 'hello, world'
print(s[0])
```
```
error: Expression [1]:17:12: 'StringLiteral' is not subscriptable, it does not implement the `__getitem__`/`__setitem__` methods
    print(s[0])
          ~^
```

Similar errors for doing this with String or StringRef objects too.

Are there other alternatives available at the moment to achieve the equivalent? Or someway to convert the String into an array/vector of underlying bytes?

**Comment by TZDak** [comment]
That should work. Probably just not implemented yet. If you use the Python.str type for now, I would think you could work around that.
Perhaps something like...
```
let s = 'hello, world'
print(Python.str(s)[0])
```

**Reply by abhinav-upadhyay** [reply]
That also doesn't appear to work. Did it work for you?

**Comment by czheo** [comment]
Try this
``` python
from String import String
print(String("hello")[2])
```

**Reply by abhinav-upadhyay** [reply, marked as answer]
Thanks, I had tried this last week. Looks like in the latest release they added support for this.
